client-test/tests/client.rs

```diff
@@ -51,12 +51,10 @@ use {
     tungstenite::connect,
 };
 
-static NEXT_RPC_PUBSUB_PORT: AtomicU16 = AtomicU16::new(rpc_port::DEFAULT_RPC_PUBSUB_PORT);
-
 fn pubsub_addr() -> SocketAddr {
     SocketAddr::new(
         IpAddr::V4(Ipv4Addr::UNSPECIFIED),
-        NEXT_RPC_PUBSUB_PORT.fetch_add(1, Ordering::Relaxed),
+        rand::thread_rng().gen_range(1024..65535),
```

Comment:
this is to prevent conflicts in tests running in parallel?

Reply:
yes. static variables don't work as expected here 😢

Reply:
yeah the statics will only be valid for tests with the same memory space, i think only same module test will share it...

if we still get conflicts (i'd expect them to be much less frequent), then we can sample for an open one instead of pure random